ingestion/src/main/java/feast/ingestion/transform/specs/WriteFeatureSetSpecAck.java

```diff
@@ -17,29 +17,38 @@
 package feast.ingestion.transform.specs;
 
 import com.google.auto.value.AutoValue;
-import feast.proto.core.FeatureSetProto;
+import feast.common.models.FeatureSetReference;
 import feast.proto.core.IngestionJobProto;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import org.apache.beam.sdk.io.kafka.KafkaIO;
-import org.apache.beam.sdk.transforms.DoFn;
-import org.apache.beam.sdk.transforms.PTransform;
-import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.*;
+import org.apache.beam.sdk.transforms.windowing.AfterPane;
+import org.apache.beam.sdk.transforms.windowing.GlobalWindows;
+import org.apache.beam.sdk.transforms.windowing.Repeatedly;
+import org.apache.beam.sdk.transforms.windowing.Window;
 import org.apache.beam.sdk.values.KV;
 import org.apache.beam.sdk.values.PCollection;
 import org.apache.beam.sdk.values.PDone;
 import org.apache.kafka.common.serialization.ByteArraySerializer;
 import org.apache.kafka.common.serialization.StringSerializer;
+import org.joda.time.Duration;
 
 /**
- * Converts input {@link feast.proto.core.FeatureSetProto.FeatureSetSpec} into {@link
+ * Collects output from sinks prepareWrite (several streams flatten into one). As soon as count of
+ * each FeatureSetReference reach getSinksCount() - it means that enough amount of sinks updated its
+ * state - ack is pushed.
```

Comment:
What happens if that getSinksCount() is not reached? (in case of failure)

Reply:
Spec wouldn't be acknowledged. FeatureSet would remain in PENDING state.
Issue must be fixed and Job must be restarted - it will reread specs and acknowledge. Currently all issues are gonna be related to BigQuery: either bq unavailable (very low probability) or we couldn't merge schema (so it makes sense not to ack such spec).